### PR TITLE
 BIGINT the CALL_CACHING_HASH_ENTRY PK [develop edition]

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,14 @@
 
 ## 43 Release Notes
 
+### Call caching database refactoring
+
+Cromwell's `CALL_CACHING_HASH_ENTRY` primary key has been refactored to use a `BIGINT` datatype in place of the previous
+`INT` datatype. Cromwell will not be usable during the time the Liquibase migration for this refactor is running.
+In the Google Cloud SQL environment this migration runs at a rate of approximately 100,000 `CALL_CACHING_HASH_ENTRY`
+rows per second. In deployments with millions or billions of `CALL_CACHING_HASH_ENTRY` rows the migration may require  
+a significant amount of downtime so please plan accordingly.
```
Comment (cjllanwarne) [inline CHANGELOG.md]: Any advice on the best way to calculate the table size? My naive `select count(*) from ...` often turns out to not be a very good idea

File: docs/developers/Arch.md [hunk 2]
```diff
@@ -55,19 +55,19 @@
 
 | Field                      | Type         | Null | Key | Default | Extra          |
 |----------------------------|--------------|------|-----|---------|----------------|
-| CALL_CACHING_HASH_ENTRY_ID | int(11)      | NO   | PRI | NULL    | auto_increment |
+| CALL_CACHING_HASH_ENTRY_ID | bigint(20)   | NO   | PRI | NULL    | auto_increment |
```
Comment (cjllanwarne) [inline docs/developers/Arch.md]: TOL: This documentation seems strangely specific when it comes to:
* mysql-specific table structure

And strangely non-specific when it comes to:
* the actual table names
* what the tables are doing
* which fields are FKs
  * ... and which fields in which other tables those FKs link to

File: CHANGELOG.md (hunk 1 above)
Comment (aednichols) [reply CHANGELOG.md]: I use
```
SELECT * FROM information_schema.innodb_sys_tablestats;
```
which is instant

Comment (mcovarr) [reply CHANGELOG.md]: looks like one needs to be root for that though

File: docs/developers/Arch.md (hunk 2 above)
Comment (mcovarr) [reply docs/developers/Arch.md]: Yeah this kind of surprised me too but I only went in here to adjust the PK datatype and fix a bogus reference to the PK.

**Pull request merged**